### PR TITLE
Add Slider on MRTK/Examples MaterialGallery based scene

### PR DESCRIPTION
**Pull request by HoloMoto**
## Overview
Based on the MaterialGallary scene of MRTK Examples, I created a feature for developers to better understand the mechanism of MRTK StandardShader.

## Changes
Fixes: #6724 

A new feature folder was created in Assets / MixedRealityToolkt.Examples / Demos / StandardShader and new features and tests were created.

ezgif com-crop
20191128_094508_HoloLens_TrimB

20191128_094508_HoloLens_TrimC

## Video
https://youtu.be/tvoDa4lt2s0

## Verification
I'm a beginner so there may be problems.

### CODE REVIEW

**Comment by cre8ivepark** [comment]
@Sthesia HoloMoto-san, please delete all files under 'app' folder, commit, and push. They are Unity generated files when you build UWP apps.

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/EmotionalButton.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class EmotionalButton : MonoBehaviour
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/EmotionalButton.cs]: can you provide a class description of what this button class does?

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/EmotionalButton.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+{
+    public GameObject ActiveOnOff;
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/EmotionalButton.cs]: please add a comment describing the public fields of your class

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/EmotionalButton.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    private bool act;
+
+
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/EmotionalButton.cs]: remove excess whitespace

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/MaterialParameterCecker.cs [hunk 4]
```diff
@@ -0,0 +1,374 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+using UnityEngine.UI;
+using Microsoft.MixedReality.Toolkit.UI;
+
+
+public class MaterialParameterCecker : MonoBehaviour
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/MaterialParameterCecker.cs]: please provide a class description with /// comment so users can read up on it in the API documentation

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/MaterialParameterCecker.cs [hunk 5]
```diff
@@ -0,0 +1,374 @@
...
+{
+    //TargetMaterial
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/MaterialParameterCecker.cs]: add /// comment documentation to all public fields and methods for API documentation

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/MaterialParameterCecker.cs [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+    public Material material;
+    //SliderValueText
+    public Text  SliderValueText;
+    public List<Text> SliderValueTextList;
+
+    //HoverColor
+    private float hoverRed;
+    private float hoverGreen;
+    private float hoverBule;
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/MaterialParameterCecker.cs]: typo: hoverBlue

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/MaterialParameterCecker.cs [hunk 7]
```diff
@@ -0,0 +1,374 @@
...
+
+    // RimLightColor
+    private float RimcolorRed;
+    private float RimcolorGreen;
+    private float RimcolorBlue;
+
+    private void SetText(float fvalue)
+    {
+        
+                SliderValueText.text = fvalue.ToString();
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/MaterialParameterCecker.cs]: remove whitespace and fix wrong intendation

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/MaterialParameterCecker.cs [hunk 8]
```diff
@@ -0,0 +1,374 @@
...
+            textList[1].text = xGvalue.ToString();
+            textList[2].text = xBvalue.ToString();
+            textList[3].text = yRvalue.ToString();
+            textList[4].text = yGvalue.ToString();
+            textList[5].text = yBvalue.ToString();
+            textList[6].text = zRvalue.ToString();
+            textList[7].text = zGvalue.ToString();
+            textList[8].text = zBvalue.ToString();
+        }
+
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/MaterialParameterCecker.cs]: remove whitespace

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/MaterialParameterCecker.cs [hunk 9]
```diff
@@ -0,0 +1,374 @@
...
+    {
+        
+                SliderValueText.text = fvalue.ToString();
+        Debug.Log(SliderValueText.text);
+        Debug.Log(fvalue);
+    }
+
+    private void SetColorText(List<Text> textList,float xRvalue,float xGvalue, float xBvalue,float yRvalue, float yGvalue, float yBvalue,float zRvalue, float zGvalue, float zBvalue)
+    {
+        if (textList.Count >= 3)
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/MaterialParameterCecker.cs]: I think this should be: if (textList.Count >= 8) 
else you might access an element out of the list bounds

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/MaterialParameterCecker.cs [hunk 10]
```diff
@@ -0,0 +1,374 @@
...
+    //InnerGlowColor
+    public void InnerGlowColor()
+    {
+        Color color = new Color(InnerGlowRed, InnerGlowGreen, InnerGlowBlue);
+        this.material.SetColor("_InnerGlowColor", color);
+    }
+    //InnerGlowPower
+    public void OnSliderUpdatedColor_InnerGlowPower(SliderEventData eventData)
+    {
+        this.material.SetFloat("_InnerGlowPower", (eventData.NewValue*30f+2f));
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/MaterialParameterCecker.cs]: why is it *30 +2f , can you explain the magic numbers with a comment?

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/MaterialParameterCecker.cs [hunk 11]
```diff
@@ -0,0 +1,374 @@
...
+    //IridescenceThresholdThreshold
+    public void OnSliderUpdatedColor_IridescenceThreshold(SliderEventData eventData)
+    {
+        this.material.SetFloat("_IridescenceThreshold", eventData.NewValue);
+        SetText( eventData.NewValue);
+    }
+    //IridescenceAngle
+    public void OnSliderUpdatedColor_IridescenceAngle(SliderEventData eventData)
+    {
+        this.material.SetFloat("_IridescenceAngle",( eventData.NewValue*1.56f - 0.78f));
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.Examples/Demos/StandardShader/New feature/MaterialParameterCecker.cs]: please explain why *1.56f - 0.78f in a comment

**Comment by cre8ivepark** [comment]
Scripts are missing on the objects. Could you please check?
2019-11-29 10_15_35-MRTK-Public-Microsoft - TestScene - Universal Windows Platform - Unity 2019 2 13
2019-11-29 10_15_17-Inbox - DongYoon Park@microsoft com - Outlook

**Comment by cre8ivepark** [comment]
Please revert these files:
Packages/manifest.json
ProjectSettings/ProjectSettings.asset
ProjectSettings/ProjectVersion.txt

**Comment by cre8ivepark** [comment]
Hi @Sthesia, I pushed some of the updates:
- Renamed the scene as 'ShaderParameterExamples.unity'. 
- Removed unnecessary script 'EmotionalButton.cs' Since this is simply showing/hiding objects, used HoloLens 2 button prefab's OnClick() event and GameObject.SetActive() to show/hide objects.
- Added Description panel. 
- Replaced HoloLens 1 style buttons with HoloLens 2 style buttons. 
2019-12-02 14_27_10-Unity 2018 4 13f1 Personal -  PREVIEW PACKAGES IN USE  - ShaderParameterExamples

![Uploading 2019-12-02 14_26_51-Unity 2018.4.13f1 Personal - [PREVIEW PACKAGES IN USE] - ShaderParameterExamples.png…]()

**Comment by cre8ivepark** [comment]
@Sthesia Could you please check Unity UI Text(which displays the slider value)? It is broken. It might be better to use TextMesh Pro. See the slider with text examples in the **SliderExamples**.unity scene.

**Comment by Sthesia** [comment]
@cre8ivepark  
> Could you please check Unity UI Text(which displays the slider value)? It is broken.
understood. I`ll check.
>Does this scene require material changes under this folder? I see many changes in the material files . Assets/MixedRealityToolkit.Examples/Demos/StandardShader/Materials/

No, no need
Perhaps the test scene has been modified because it brought objects from the Material gallery.

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/Scripts/MaterialParameterChanger.cs [hunk 12]
```diff
@@ -0,0 +1,532 @@
...
+
+namespace Microsoft.MixedReality.Toolkit.Examples
+{
+    /// <summary>
+    /// This class changes the MRTKStandardShader parameter of the material attached using PinchSlider.
+    /// By acquiring each method in the event of PinchSlider.sc, you can see the function of MRTKStandardShader on the actual machine.
+    /// </summary>
+    public class MaterialParameterChanger : MonoBehaviour
+    {
+        /// <summary>
```
Comment (Troy-Ferrell) [inline Assets/MixedRealityToolkit.Examples/Demos/StandardShader/Scripts/MaterialParameterChanger.cs]: Can you remove these comments per our coding guidelines?
https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html
"comments that restate what is obvious provide no extra information and increase the noise to signal ratio."

Comment (Troy-Ferrell) [reply Assets/MixedRealityToolkit.Examples/Demos/StandardShader/Scripts/MaterialParameterChanger.cs]: Same with the items below

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/Scripts/MaterialParameterChanger.cs [hunk 13]
```diff
@@ -0,0 +1,532 @@
...
+        /// <param name="xRvalue">The first parameter set in the text list</param>
+        /// <param name="xGvalue">The second parameter set in the text list</param>
+        /// <param name="xBvalue">The third parameter set in the text list</param>
+        /// <param name="yRvalue">The fourth parameter set in the text list</param>
+        /// <param name="yGvalue">The fifth parameter set in the text list</param>
+        /// <param name="yBvalue">The sixth parameter set in the text list</param>
+        /// <param name="zRvalue">The seventh parameter set in the text list</param>
+        /// <param name="zGvalue">The eighth parameter set in the text list</param>
+        /// <param name="zBvalue">The ninth parameter set in the text list</param>
+        private void SetText(List<Text> textList, float xRvalue, float xGvalue, float xBvalue, float yRvalue, float yGvalue, float yBvalue, float zRvalue, float zGvalue, float zBvalue)
```
Comment (Troy-Ferrell) [inline Assets/MixedRealityToolkit.Examples/Demos/StandardShader/Scripts/MaterialParameterChanger.cs]: Looking through the example scene. I'm not seeing this used anywhere.

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/Scripts/MaterialParameterChanger.cs [hunk 14]
```diff
@@ -0,0 +1,532 @@
...
+        private float somootheness;
+        private float roundcornermargin;
+        private float roundcornerradius;
+
+        private float hoverRed;
+        private float hoverGreen;
+        private float hoverBlue;
+
+        private float InnerGlowRed;
+        private float InnerGlowGreen;
```
Comment (Troy-Ferrell) [inline Assets/MixedRealityToolkit.Examples/Demos/StandardShader/Scripts/MaterialParameterChanger.cs]: I don't believe these are all necessary? They aren't exactly accurate as every instance of the component is duplicating these and some are populated while others aren't.

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/Scripts/MaterialParameterChanger.cs [hunk 15]
```diff
@@ -0,0 +1,532 @@
...
+        /// <param name="yRvalue">The fourth parameter set in the text list</param>
+        /// <param name="yGvalue">The fifth parameter set in the text list</param>
+        /// <param name="yBvalue">The sixth parameter set in the text list</param>
+        /// <param name="zRvalue">The seventh parameter set in the text list</param>
+        /// <param name="zGvalue">The eighth parameter set in the text list</param>
+        /// <param name="zBvalue">The ninth parameter set in the text list</param>
+        private void SetText(List<Text> textList, float xRvalue, float xGvalue, float xBvalue, float yRvalue, float yGvalue, float yBvalue, float zRvalue, float zGvalue, float zBvalue)
+        {
+            x = textList.Count;
+            Debug.Log(textList.Count);
```
Comment (Troy-Ferrell) [inline Assets/MixedRealityToolkit.Examples/Demos/StandardShader/Scripts/MaterialParameterChanger.cs]: These can be removed

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/Scripts/MaterialParameterChanger.cs (hunk 13 above)
Comment (Troy-Ferrell) [reply Assets/MixedRealityToolkit.Examples/Demos/StandardShader/Scripts/MaterialParameterChanger.cs]: It's also very confusing to read and understand for new users. Some assumptions and types built in here.

File: Assets/MixedRealityToolkit.Examples/Demos/StandardShader/Scripts/MaterialParameterChanger.cs (hunk 12 above)
Comment (Sthesia) [reply Assets/MixedRealityToolkit.Examples/Demos/StandardShader/Scripts/MaterialParameterChanger.cs]: I'll try.

**Pull request closed without merging**